### PR TITLE
chore: Drop node lease garbage collection controller

### PR DESCRIPTION
**Pull request by jigisha620**
Fixes #N/A 

**Description**
Drop node lease garbage collection controller since this is now taken care by the change we made to wait for instance termination before deleting the nodeClaim.

**How was this change tested?**

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/controllers/controllers.go [hunk 1]
```diff
@@ -90,7 +89,6 @@ func NewControllers(
 		nodeclaimgarbagecollection.NewController(clock, kubeClient, cloudProvider),
 		nodeclaimtermination.NewController(kubeClient, cloudProvider, recorder),
 		nodeclaimdisruption.NewController(clock, kubeClient, cloudProvider),
-		leasegarbagecollection.NewController(kubeClient),
```
Comment (jonathan-innis) [inline pkg/controllers/controllers.go:93]: Do we have metrics in place and monitoring to ensure that when we remove this that we are not leaking node leases now?

**Review by jonathan-innis — COMMENTED** [review]
/lgtm
/approve

File: pkg/controllers/controllers.go (hunk 1 above)
Comment (jigisha620) [reply pkg/controllers/controllers.go:93]: Added a test to monitor leaked leases - https://github.com/aws/karpenter-provider-aws/pull/6829

**Review by jigisha620 — COMMENTED** [review]
/unhold

**Pull request merged**